tests/test_history.py

```diff
@@ -63,26 +63,62 @@ def hist():
     return h
 
 def test_history_class_span(hist):
-    h = hist
-    assert h == ['first', 'second', 'third', 'fourth']
-    assert h.span('-2..') == ['third', 'fourth']
-    assert h.span('2..3') == ['second', 'third']    # Inclusive of end
-    assert h.span('3') == ['third']
-    assert h.span(':') == h
-    assert h.span('2..') == ['second', 'third', 'fourth']
-    assert h.span('-1') == ['fourth']
-    assert h.span('-2..-3') == ['third', 'second']
-    assert h.span('*') == h
+    for tryit in ['*', ':', '-', 'all', 'ALL']:
+        assert hist.span(tryit) == hist
+
+    assert hist.span('3') == ['third']
+    assert hist.span('-1') == ['fourth']
+
+    assert hist.span('2..') == ['second', 'third', 'fourth']
+    assert hist.span('2:') == ['second', 'third', 'fourth']
+
+    assert hist.span('-2..') == ['third', 'fourth']
+    assert hist.span('-2:') == ['third', 'fourth']
+
+    assert hist.span('1..3') == ['first', 'second', 'third']
+    assert hist.span('1:3') == ['first', 'second', 'third']
+
+    assert hist.span(':-2') == ['first', 'second']
```

Comment:
This unit test works as intended and as I would expect, however when I use it with the history command in a ``cmd2`` app, it appears to treat the trailing negative index as exclusive whereas as positive one is treated as inclusive.